src/tainting/Dataflow_tainting.ml

```diff
@@ -367,6 +367,31 @@ let sink_biased_union_mvars source_mvars sink_mvars =
   in
   Some (source_mvars' @ sink_mvars)
 
+let merge_source_mvars taints_and_bindings =
```

Comment:
I think this could end up picking an arbitrary `$X` among the sources. If there are sources A, B, and C, each one binding `$X` to something else. First `merge A B` will remove `$X`, and then we  `merge (merge A B) C` and C will add `$X` again.

Reply:
good catch! i reimplemented + tested, and behavior is the same (as well as working for the 3-source case now)